### DISCUSSION TITLE
Value menjadi NaN ketika diterapkan method reindex pada 'data series'

### DISCUSSION THREAD

**Discussion by FrengkyLR** [teamalgoritma/community#33] [Q&A] [answered]
pertanyaan : Value menjadi NaN ketika diterapkan method reindex
Error :value yang semula ada menjadi NaN

Code:
x = pd.date_range(start="2021-01-02", end='2021-03-29')
close_price = close_price.reindex(x)
close_price.head()
capt:
![Capture](https://user-images.githubusercontent.com/101881998/160730129-9465ca36-45d5-424d-a3c9-52c462b9bead.PNG)

**Comment by tomytjandra** [comment]
Code `pd.date_range(start="2021-01-02", end='2021-03-29')` digunakan untuk membuat daftar tanggal harian dari tanggal 2 Januari 2021 sampai 29 Maret 2021. Setelah reindex diterapkan, value menjadi NaN ketika di object `close_price` pada tanggal tersebut tidak ada nilainya.

Maka dari itu, pastikan pada object `close_price` terdapat nilai pada tanggal 2 Januari 2021 sampai 29 Maret 2021. Apabila ternyata ada, mohon lampirkan juga code yang Anda gunakan untuk menarik data saham. Lampirkan juga dataframe `close_price` sebelum dilakukan reindex dalam bentuk json dengan method `.to_json()`.

Note: Mohon untuk mengikuti [guideline berikut](https://github.com/teamalgoritma/community/discussions/29#discussion-3965233) agar pertanyaan Anda dapat kami jawab dengan lebih rinci. Terima kasih.

**Reply by FrengkyLR** [reply]
Berikut Kak lampiran code penarikan data saham dan data_frame untuk close_price pada tanggal tersebut

Code Penarikan Data Saham :

![1](https://user-images.githubusercontent.com/101881998/160741606-0617565f-fcc9-4889-a993-4d4b73beb0ad.PNG)

Dataframe Close_Price ( 2 Januari 2021 sampai 29 Maret 2021 ) :

![22](https://user-images.githubusercontent.com/101881998/160741619-5ab6341c-9515-48e1-86ea-9cdd46fdbbe1.PNG)

**Comment by dwiknrd** [comment]
Halo [FrengkyLR](https://github.com/FrengkyLR), 

Seharusnya kode tersebut sudah benar, kami juga sudah mencoba kode tersebut pada local kami dan dataframe yang dihasilkan tidak menjadi nilai NaN semua.

Berikut adalah kode yang kami coba jalankan:

```
import yfinance as data

symbol = ['TSLA', 'TM']
start_date = '2018-01-01' # 1 Januari 2018
end_date = '2022-03-30' # 30 Maret 2022
stock = data.download(symbol, start_date, end_date)
stock.columns.names = ['Status', 'Kode']
stock.head()

close_price = stock['Close'].loc["2021-01-02":"2021-03-29",:]
x = pd.date_range(start="2021-01-02", end='2021-03-29')
close_price = close_price.reindex(x)
close_price.head()
```

Pastikan, tidak melakukan running kode terus `date_range()` secara terus-menerus karena dikhawatirkan dapat menyebabkan nilai pada dataframe menjadi NaN.

Semoga membantu 😀

**Comment by yosiaazarya** [comment, marked as answer]
Pastikan index yang Bapak gunakan sudah berbentuk `datetime64`. Untuk memastikan hal tersebut Bapak dapat menjalankan code berikut sebelum melakukan `.reindex()` : `close_price.index =  pd.to_datetime(close_price.index , format='%Y-%m-%d')` . Demikian jawaban yang mungkin dapat membantu. 😃

**Comment by FrengkyLR** [comment]
Terimakasih sudah solved, sangat membantu
